.github/workflows/trigger-redeploy.yml

```
@@ -8,9 +8,127 @@ on:
         required: false
         default: 'Manual trigger'
 
+  push:
+    paths:
+      - 'scripts/openapi-capitalize-language.mjs'
+      - 'scripts/openapi-split-break-circular.mjs'
+    branches:
+      - main
+
+permissions:
+  contents: write
+  pull-requests: write
+
 jobs:
   trigger-redeploy:
     runs-on: ubuntu-latest
+    outputs:
+      has_changes: ${{ steps.check-generated.outputs.has_changes }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set up mise
+        uses: jdx/mise-action@v2
+        with:
+          cache: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install dependencies
+        run: yarn install --immutable
+
+      - name: Regenerate OpenAPI documentation
+        run: |
+          echo "📋 Regenerating OpenAPI documentation..."
+          echo "Reason: ${{ github.event.inputs.reason }}"
+          yarn openapi:regenerate:all
+          echo "✅ Documentation regenerated successfully"
+
+      - name: Check for generated changes
+        id: check-generated
+        run: |
+          if [[ -n $(git status --porcelain) ]]; then
+            echo "has_changes=true" >> $GITHUB_OUTPUT
+            echo "📝 Changes detected in generated documentation"
+          else
+            echo "has_changes=false" >> $GITHUB_OUTPUT
+            echo "✅ No changes in generated documentation"
+          fi
+
+      - name: Create Pull Request
+        if: steps.check-generated.outputs.has_changes == 'true'
+        id: create-pr
+        uses: peter-evans/create-pull-request@v5
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: 'chore: regenerate OpenAPI documentation'
+          title: '[Manual] Regenerate OpenAPI Documentation'
+          body: |
+            ## 🔄 Manual OpenAPI Documentation Regeneration
+
+            This PR was manually triggered to regenerate the OpenAPI documentation.
+
+            **Trigger reason**: ${{ github.event.inputs.reason }}
+            **Triggered by**: ${{ github.actor }}
+            **Run ID**: ${{ github.run_id }}
```

1/5 (nit, non-blocking)

Let's make this a link, something like:

```suggestion
            **Run ID**: [${{ github.run_id }}](https://github.com/gleanwork/glean-developer-site/actions/runs/${{ github.run_id }})
```